src/containers/Resources/index.jsx

```diff
@@ -1,18 +1,114 @@
+    .catch((error) => {
+      alert(`Error getting documents ===> ${error.message}`);
+    });
+};
+
+const checkIfIsIncluded = (option, ngoValue) => {
+  if (!option.includes(ngoValue)) {
+    option.push(ngoValue);
+  }
+};
+
+// Advance Filters component's functionalities - START
+// OUTPUT ==> [{city:["ankara","Sincan"]}, {service: ["health care", "finance"]}]
+const getFilterOptions = (ngos) => {
+  const filterOptions = { cities: [], services: [], ratings: [], names: [] };
+
+  for (let i = 0; i < ngos.length; i++) {
+    checkIfIsIncluded(filterOptions.cities, ngos[i].city);
+    checkIfIsIncluded(filterOptions.services, ngos[i].service);
+    checkIfIsIncluded(filterOptions.ratings, ngos[i].rating);
+    checkIfIsIncluded(filterOptions.names, ngos[i].name);
+    return filterOptions;
+  }
+};
+
+const applyAdvancedFilters = (filterOptions, ngos, setFilteredNgos) => {
+  setFilteredNgos([]);
+  for (let i = 0; i < ngos.length; i++) {
+    const isAllOptionsIncluded =
+      filterOptions.cities.includes(ngos[i].city) &&
+      filterOptions.services.includes(ngos[i].service) &&
+      filterOptions.names.includes(ngos[i].name) &&
+      filterOptions.ratings.includes(ngos[i].rating);
+
+    if (isAllOptionsIncluded) {
+      setFilteredNgos((prevState) => [...prevState, ngos[i]]);
+    }
+  }
+};
+// Advance Filters component's functionalities - END
+
+// Search bar component's functionalities - START
+const isSearchWordsInNgo = (ngo, searchedWords) => {
+  const searchedWordsArray = searchedWords.split(" ");
+  const searchedValue =
+    searchedWordsArray.length > 0
+      ? new RegExp(searchedWordsArray.join("|"), "i")
+      : new RegExp(searchedWordsArray[0], "i");
+  return Object.keys(ngo).some((parameter) =>
+    searchedValue.test(ngo[parameter])
+  );
+};
+
+const filterNgos = (ngos, searchedWords) => {
+  return ngos.filter((ngo) => isSearchWordsInNgo(ngo, searchedWords));
+};
+// Search bar component's functionalities - END
+
 const Resources = () => {
+  const [allNgos, setAllNgos] = useState([]);
+  useEffect(() => fetchNgos(setAllNgos), []);
+
+  const [filterOptions, setFilterOptions] = useState([]);
+  setFilterOptions(getFilterOptions(allNgos));
+
+  const [filteredNgos, setFilteredNgos] = useState([]);
+
+  const [searchKey, setSearchKey] = useState("");
+  const handleSearchSubmit = () => {
+    setFilteredNgos(filterNgos(allNgos, searchKey));
+    setSearchKey("");
+  };
+
   return (
     <div>
-      <SearchBar />
+      <SearchBar
+        searchKey={searchKey}
+        setSearchKey={setSearchKey}
+        handleSearchSubmit={handleSearchSubmit}
+      />
       <div className="d-flex">
-        <AdvancedFilters />
-        <NgoList />
+        <AdvancedFilters
+          applyAdvancedFilters={() =>
+            applyAdvancedFilters(filterOptions, allNgos, setFilteredNgos)
+          }
+          clearAll={setFilteredNgos}
+        />
+        <NgoList ngos={filteredNgos !== undefined ? filteredNgos : allNgos} />
```

Comment:
here u can do filteredNgos ? filteredNgos: AllNgos
since the question mark already checks if the variable is true or flase